cmd/juju/model/exportbundle_test.go

```diff
@@ -0,0 +1,165 @@
+		ModelUUID: testing.ModelTag.Id(),
+		ModelType: coremodel.IAAS,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	s.store.Models["testing"].CurrentModel = "admin/mymodel"
+}
+
+func (s *ExportBundleCommandSuite) TestExportBundleFailOnv1(c *gc.C) {
+	s.fake.result = ""
+	s.fake.Stub.SetErrors(errors.New("command not supported on v1"))
+	s.fake.bestAPIVersion = 1
+
+	ctx, err := cmdtesting.RunCommand(c, model.NewExportBundleCommandForTest(s.fake, s.store))
+	c.Assert(err, gc.NotNil)
+
+	s.fake.CheckCalls(c, []jujutesting.StubCall{
+		{"ExportBundle", nil},
+	})
+
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, gc.Equals, "")
+	c.Assert(err, gc.ErrorMatches, "command not supported on v1")
+}
+
+func (s *ExportBundleCommandSuite) TestExportBundleFailEmptyResult(c *gc.C) {
+	s.fake.result = ""
+	s.fake.Stub.SetErrors(errors.New("export failed: nothing to export as there are no applications"))
+	s.fake.bestAPIVersion = 2
+
+	ctx, err := cmdtesting.RunCommand(c, model.NewExportBundleCommandForTest(s.fake, s.store))
+	c.Assert(err, gc.NotNil)
+
+	s.fake.CheckCalls(c, []jujutesting.StubCall{
+		{"ExportBundle", nil},
+	})
+
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, gc.Equals, "")
+	c.Assert(err, gc.ErrorMatches, "export failed: nothing to export as there are no applications")
+}
+
+func (s *ExportBundleCommandSuite) TestExportBundleSuccessNoFilename(c *gc.C) {
+	s.fake.result = "applications:\n" +
+		"  mysql:\n" +
+		"    charm: \"\"\n" +
+		"    num_units: 1\n" +
+		"    to:\n" +
+		"    - \"0\"\n" +
+		"  wordpress:\n" +
+		"    charm: \"\"\n" +
+		"    num_units: 2\n" +
+		"    to:\n" +
+		"    - \"0\"\n" +
+		"    - \"1\"\n" +
+		"machines:\n" +
+		"  \"0\": {}\n" +
+		"  \"1\": {}\n" +
+		"series: xenial\n" +
+		"relations:\n" +
+		"- - wordpress:db\n" +
+		"  - mysql:mysql\n"
+	s.fake.bestAPIVersion = 2
+
+	ctx, err := cmdtesting.RunCommand(c, model.NewExportBundleCommandForTest(s.fake, s.store))
+	c.Assert(err, jc.ErrorIsNil)
+	s.fake.CheckCalls(c, []jujutesting.StubCall{
+		{"ExportBundle", nil},
+	})
+
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, gc.Equals, "|\n"+
+		"  applications:\n"+
+		"    mysql:\n"+
+		"      charm: \"\"\n"+
+		"      num_units: 1\n"+
+		"      to:\n"+
+		"      - \"0\"\n"+
+		"    wordpress:\n"+
+		"      charm: \"\"\n"+
+		"      num_units: 2\n"+
+		"      to:\n"+
+		"      - \"0\"\n"+
+		"      - \"1\"\n"+
+		"  machines:\n"+
+		"    \"0\": {}\n"+
+		"    \"1\": {}\n"+
+		"  series: xenial\n"+
+		"  relations:\n"+
+		"  - - wordpress:db\n"+
+		"    - mysql:mysql\n")
+}
+
+func (s *ExportBundleCommandSuite) TestExportBundleSuccessFilename(c *gc.C) {
+	s.fake.bestAPIVersion = 2
+
+	ctx, err := cmdtesting.RunCommand(c, model.NewExportBundleCommandForTest(s.fake, s.store), "--filename", "mymodel")
+	c.Assert(err, jc.ErrorIsNil)
+	s.fake.CheckCalls(c, []jujutesting.StubCall{
+		{"ExportBundle", nil},
+	})
+
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, gc.Equals, "Bundle successfully exported to mymodel.yaml\n")
```

Comment:
You need to check the contents of the file :)

Reply:
done

Reply:
I am pretty sure that we do not want to leave behind the files created by tests.  You would need to have a TearDownTest os.Remove'ing the file.. There are some examples around, especially on the commands that do use files like "create-backup", etc.

Reply:
done